### PR TITLE
Fully support addresses without ports

### PR DESCRIPTION
**Pull request by prashanthpai**
* Allow addresses without ports during add peer operation
* Normalize peer address before comparison

### CODE REVIEW

**Comment by prashanthpai** [comment]
@kshlm This fixes the glitch that occurred during the demo earlier for add peer. Need to fix the absence of port elsewhere too (like in address specified during volume create)

File: utils/peerutils.go [hunk 1]
```diff
@@ -3,27 +3,38 @@ package utils
...
 // return an address of the form <ip:port>
 func FormRemotePeerAddress(peeraddress string) (string, error) {
-	var remotePeerAddress string
 
 	host, port, err := net.SplitHostPort(peeraddress)
-	if err != nil || host == "" {
-		return "", errors.New("Invalid peer address")
+	if err != nil {
+		// net.SplitHostPort() returns an error if port is missing.
+		if strings.HasPrefix(err.Error(), "missing port in address") {
```
Comment (kshlm) [inline utils/peerutils.go]: Did you try to find an alternate way to check for error here? There should be a better way to check for this error other than comparing strings.

Comment (prashanthpai) [inline utils/peerutils.go]: I did this after looking at the [source code](https://golang.org/src/net/ipsock.go?s=3288:3354#L99) of `SplitHostPort`. The alternative was to use `strings.Split`

Comment (kshlm) [inline utils/peerutils.go]: Okay then. I'll merge. But this just feels not right.

**Pull request merged**